aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -703,16 +697,13 @@ module aptos_framework::stake {
         assert!(exists<ValidatorConfig>(pool_address), error::not_found(EVALIDATOR_CONFIG));
         let validator_info = borrow_global_mut<ValidatorConfig>(pool_address);
         let old_consensus_pubkey = validator_info.consensus_pubkey;
-        // Checks the public key has a valid proof-of-possession to prevent rogue-key attacks.
+        // Checks the public key is valid to prevent rogue-key attacks.
         if (!genesis) {
-            let pubkey_from_pop = &mut bls12381::public_key_from_bytes_with_pop(
-                new_consensus_pubkey,
-                &proof_of_possession_from_bytes(proof_of_possession)
-            );
-            assert!(option::is_some(pubkey_from_pop), error::invalid_argument(EINVALID_PUBLIC_KEY));
+            let validated_public_key = ed25519::new_validated_public_key_from_bytes(new_consensus_pubkey);
+            assert!(option::is_some(&validated_public_key), error::invalid_argument(EINVALID_PUBLIC_KEY));
```

Comment:
The branch on `!genesis` is not at all required since you are doing the same thing in both the branch now.

Reply:
Done https://github.com/Entropy-Foundation/aptos-core/pull/9/commits/3fefc82a2c107f84b89b654ccb012cbdd6c7d3a1